### PR TITLE
API keys: Disabling of API keys endpoints

### PR DESCRIPTION
**Pull request by eleijonmarck**
**What is this feature?**

- This PR removes the API key endpoints from Grafana API keys in favor of Service account tokens. 
- Deprecated the resource for terraform in v3 - https://github.com/grafana/terraform-provider-grafana/pull/1529

**Why do we need this feature?**
This is a step of making Grafana to only use service accounts and service account tokens as part of our way to making a unified machine access experience in Grafana.

For more information see the epic: #53567.

**Which issue(s) does this PR fix?**:
#53567

# Release notice breaking change
We are disabling the API keys endpoint routes in favor of using service accounts and service accounts tokens.

**What does this mean for me?**
- Any “existing” API keys will continue to work as authentication methods
- No new APIkeys will be able to be created from POST `/api/auth/apikeys`

The migration from apikeys to service accounts is noted down in move from [api keys to service account tokens](https://grafana.com/docs/grafana/latest/administration/api-keys/#migrate-api-keys-to-grafana-service-accounts).

### CODE REVIEW

**Review by kalleep — COMMENTED** [review]
Great! You need to add breaking change label to this pr

**Review by vtorosyan — CHANGES_REQUESTED** [review]
Can we please hold on from merging this change? I think we need a bit more due diligence to understand the impact and provide guidance for users when anything breaks before cutting off the write path.

File: pkg/api/apikey.go [hunk 1]
```diff
@@ -26,48 +20,17 @@ import (
...
 //
 // Responses:
-// 200: getAPIkeyResponse
-// 401: unauthorisedError
-// 403: forbiddenError
-// 404: notFoundError
-// 500: internalServerError
+// 301: statusMovedPermanently
 func (hs *HTTPServer) GetAPIKeys(c *contextmodel.ReqContext) response.Response {
-	query := apikey.GetApiKeysQuery{OrgID: c.SignedInUser.GetOrgID(), User: c.SignedInUser, IncludeExpired: c.QueryBool("includeExpired")}
```
Comment (eleijonmarck) [inline pkg/api/apikey.go:35]: @Jguer asking you, do you think we should disable the listing of APIkeys?

**Comment by eleijonmarck** [comment]
without the terraform PR we are redirecting to service account token directly for terraform and we would get a message saying:

```tf
│ Error: status: 400, body: {"message":"Service Account ID is invalid","traceID":""}
│
│   with grafana_api_key.foo,
│   on apikeys.tf line 1, in resource "grafana_api_key" "foo":
│    1: resource "grafana_api_key" "foo" {
```

**Comment by vtorosyan** [comment]
I am still a bit hesitant to remove the POST endpoint with appropriate announcement with a good notice. 

What about having a blog post sharing timeline for full sunsetting of the API keys so that users have time for migration and the breaking change?

**Comment by eleijonmarck** [comment]
> I am still a bit hesitant to remove the POST endpoint with appropriate announcement with a good notice.
> 
> What about having a blog post sharing timeline for full sunsetting of the API keys so that users have time for migration and the breaking change?

Let's bring this up as a topic for the deprecation, also to understand where we would put this into priority.

**Comment by Jguer** [comment]
Closing as unplanned for now

**Pull request closed without merging**